tests/e2e/performance/test_fio_benchmark.py

```diff
@@ -308,19 +271,236 @@ def test_fio_workload_simple(self, ripsaw, es, interface, io_pattern):
+
+        log.debug(f"Full results is : {full_results.results}")
+
+        self.copy_es_data(es, full_results)
+        full_results.analyze_results()  # Analyze the results
+        # Writing the analyzed test results to the Elastic-Search server
+        full_results.es_write()
+        full_results.codespeed_push()  # Push results to codespeed
+        # Creating full link to the results on the ES server
+        log.info(f"The Result can be found at ; {full_results.results_link()}")
+
+    @skipif_ocs_version("<4.6")
+    @pytest.mark.parametrize(
+        argnames=["io_pattern"],
+        argvalues=[
+            pytest.param(
+                *["random"],
+                marks=pytest.mark.polarion_id("OCS-846"),
+            ),
+        ],
+    )
+    def test_fio_compressed_workload(self, es, storageclass_factory, io_pattern):
+        """
+        This is a basic fio perf test which run on compressed volume
+
+        """
+
+        ripsaw = RipSaw()
+        self.ripsaw_deploy(ripsaw)
+
+        log.info("Creating compressed pool & SC")
+        sc_obj = storageclass_factory(
+            interface=constants.CEPHBLOCKPOOL,
+            new_rbd_pool=True,
+            replica=3,
+            compression="aggressive",
+        )
+
+        sc = sc_obj.name
+        # Create fio benchmark
+        log.info("Create resource file for fio workload")
+        self.fio_cr = templating.load_yaml(
+            "ocs_ci/templates/workloads/fio/benchmark_fio_cmp.yaml"
+        )
+
+        # Saving the Original elastic-search IP and PORT - if defined in yaml
+        self.es_info_backup(es)
+
+        # Setting the data set to 40% of the total storage capacity
+        self.setting_storage_usage()
+
+        self.get_env_info()
+
+        self.fio_cr["spec"]["workload"]["args"]["storageclass"] = sc
+        self.setting_io_pattern(io_pattern)
+        fio_client_pod = self.deploy_and_wait_for_wl_to_start()
+
+        # Getting the UUID from inside the benchmark pod
+        uuid = ripsaw.get_uuid(fio_client_pod)
+        # Setting back the original elastic-search information
+        self.fio_cr["spec"]["elasticsearch"] = self.backup_es
+
+        # Initialize the results doc file.
+        full_results = self.init_full_results(FIOResultsAnalyse(uuid, self.fio_cr))
+
+        # Setting the global parameters of the test
+        full_results.add_key("io_pattern", io_pattern)
+
+        end_time = self.wait_for_wl_to_finish(fio_client_pod)
+        full_results.add_key("test_time", {"start": self.start_time, "end": end_time})
+
+        log.info("verifying compression ratio")
+        results = get_ceph_df_detail()
+        for pool in results["pools"]:
+            if "cbp-test-" in pool["name"]:
+                stor = pool["stats"]["compress_under_bytes"]
+                saved = pool["stats"]["compress_bytes_used"]
+                ratio = int(100 / (stor / saved))
+                log.info(f"The compression ratio is {ratio}%")
```

Comment:
I don't see any verification here. You are simply logging the compression ratio. Did you intend to compare this ratio with some value?

Reply:
correct.
i just add the actual verification. in case of differences i just add error message to the log, not failing the test.